openrtb_ext/request_wrapper.go

```diff
@@ -333,6 +337,14 @@ func (ue *UserExt) unmarshal(extJson json.RawMessage) error {
 		}
 	}
 
+	consentedProviderSettingsJson, hasCPSettings := ue.ext["ConsentedProvidersSettings"]
+	if hasCPSettings {
+		ue.consentedProvidersSettings = &ExtUserCPSettings{}
+		if err := json.Unmarshal(consentedProviderSettingsJson, ue.consentedProvidersSettings); err != nil {
+			return err
```

Comment:
This block isn't covered by tests, can this be covered?

Reply:
Done. Added a bunch of tests to cover this entire function and some other scenarios.